### ISSUE TITLE
`codegen` project is incomplete / non-functional

### ISSUE THREAD

**Issue by Marcono1234** [google/gson#2007]
It looks like the `codegen` project is incomplete. The only thing it does is write an empty class, see:
https://github.com/google/gson/blob/a92bbf849c99b1fc708eb3dd209af9c72110ed4d/codegen/src/main/java/com/google/gson/codegen/GeneratedTypeAdapterProcessor.java#L56-L58

For example, the output for the `Order` test class is just:
```java
package com.google.gson.codegen.functional;
final class Order$TypeAdapter {
}
```

Should the project be removed, or should its functionality be implemented eventually?

**Comment by eamonnmcmanus** [comment]
I think I would be in favour of deleting it. The code is 9 years old. It seems very half-baked, has no documentation, and has what appears to be a copy of a very old version of JavaPoet. There _might_ be interest in some sort of code-generating solution, but if so I think it would be better built with modern tools, including the latest JavaPoet.

Perhaps @inder123 wants to weigh in, though?

**Comment by inder123** [comment]
proto has users. May be best to leave it as is?

**Comment by eamonnmcmanus** [comment]
Yes, sorry, the question was about the codegen project, not the proto project.

**Comment by NBXXF** [comment]
you can use https://github.com/NBXXF/gson_plugin
